### PR TITLE
Remove particle update ghost parameter

### PR DESCRIPTION
**Pull request by gassmoeller**
I think it was a bad choice to only update ghost particles if this parameter was set. It is inconsistent with the handling of FE degrees of freedom (where entries on ghost cells are always available), and by now both the linear and quadratic least squares interpolators require the ghost particles if their limiters are active. The new interpolator in #5815 also requires ghost particles in all models. Therefore and as part of #5568 I would suggest to remove the input parameter and update ghost particles unconditionally.

Note that I could keep the parameter and only change the default, and/or assert that the parameter is set to true if we want to stay more compatible with old parameter files. I would be open for discussion about that, I just thought I would start with the most radical solution and see what opinions are out there.

### CODE REVIEW

File: contrib/utilities/update_scripts/prm_files/move_particles.py [hunk 1]
```diff
@@ -113,7 +113,15 @@ def move_particle_postprocess_parameters_back(parameters):
 
     return parameters
 
+def remove_update_ghost_particles_parameter(parameters):
+    """ Remove the parameter 'Update ghost particles'. """
 
+    # Find the particle parameters and move
```
Comment (tjhei) [inline contrib/utilities/update_scripts/prm_files/move_particles.py:119]:
```suggestion
    # Find the particle parameters and remove
```

**Review by tjhei — APPROVED** [review]
Works for me.

**Comment by gassmoeller** [comment]
Comment is fixed.

**Comment by gassmoeller** [comment]
Test has been fixed as well.

**Review by tjhei — APPROVED** [review]
If nobody else complains about the backwards-incompatible change in the next few days, I will merge.

**Comment by gassmoeller** [comment]
I decided to leave the parameter in after all. No need to break existing parameter files only because they set a parameter in the correct way. I assert now that the parameter is set to true and otherwise show a message that the parameter will be removed in the future. If we leave it like this for a year or two most parameter files will be updated already by the update script. If the tests pass this should be good to go.

File: source/particle/world.cc [hunk 2]
```diff
@@ -1234,7 +1231,7 @@ namespace aspect
                                "particle weight is recommended. Before adding the weights "
                                "of particles, each cell already carries a weight of 1000 to "
                                "account for the cost of field-based computations.");
-            prm.declare_entry ("Update ghost particles", "false",
+            prm.declare_entry ("Update ghost particles", "true",
```
Comment (tjhei) [inline source/particle/world.cc:1234]: I don't think we have formalized a way to do this yet, but would you mind putting the word "deprecated" as a comment and/or inside the documentation? This way we can search for deprecated parameters after the release.

File: doc/modules/changes/20240625_gassmoeller [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Removed: The input parameter 'Update ghost particles' has been removed.
```
Comment (tjhei) [inline doc/modules/changes/20240625_gassmoeller:1]: It is not really removed anymore, right?

**Review by tjhei — APPROVED** [review]
one more comment...

File: doc/modules/changes/20240625_gassmoeller (hunk 3 above)
Comment (gassmoeller) [reply doc/modules/changes/20240625_gassmoeller]: True, I changed it to deprecated.

**Pull request merged**